### ISSUE TITLE
Setting permissions through nodejs

### ISSUE THREAD

**Issue by rguikers** [eris-ltd/eris-db.js#24]
Hi,

Would be really nice if it is possible to set permissions in a chain through javascript. Maybe already possible, but i'm not able to find out what the transaction should look like.
This is typically which will happen server side for a new account address, so there is no local signing (for the client) involved (although very eager to have local signing implemented). In this way we could write a client onboarding application in a nodejs server app. Better than using mintx or eris-pm on the command line.

Would be happy to elaborate on the use-case through a private channel.. 
Thanks,

**Comment by rguikers** [comment]
I see this is a duplicate of #18 
Would be nice to add the permission transaction to the set described..

**Comment by NodeGuy** [comment]
Would you please write an example here of how you would set the permissions you want using `mintx` or `eris-pm` and would like to do via the JavaScript API instead?  No need to go into details about your use case here if you don't want to.

**Comment by rguikers** [comment]
Sure,

From mintx using :
```
mintx permission set_base <addres> call true --sign=true --broadcast=true
```
So from javascript i would opt for a Transaction method like
```
Transactions.prototype.setBasePerm = function (address, perm, permflag, callback)
```
for example :
```
setBasePerm(<addres>, "call", true, function(err, result){
});
```
Signed by a address with root permission off course
Could also include global permissions, but are less likely to be managed by a client (in my opinion)..

Does this answer your question?

**Comment by NodeGuy** [comment]
Yes it does, thank you.

**Comment by NodeGuy** [comment]
Status update:

The JavaScript code to make this work has been written in the [#24](https://github.com/eris-ltd/eris-db.js/tree/%2324) branch.  Sending a transaction to Eris DB results in the following error message:

`[JSON RPC 2.0 Error] Message Id: 3, (INVALID_PARAMS) - Expected Byte of 2 but got 20`

Ethan suggested changing [this line](https://github.com/eris-ltd/eris-db/blob/9c0f12814b1c5b2ac9210900eb3d144b4f593028/erisdb/methods.go#L384) of Eris DB to:

```Go
var param types.Tx
```

That appears to fix the error and then the server response with HTTP status code `500` and no other information.

Solving this is going to require delving into Eris DB so I'm putting it on the back burner for now.

**Comment by rguikers** [comment]
Thanks for the update. Looking forward to have this implemented!

**Comment by NodeGuy** [comment]
It's been decided to use the Tendermint API instead of the Eris DB API for this feature.  Testing this properly will require use of the CLI so this issue is now blocked by [the integration tests passing](https://github.com/eris-ltd/eris-db.js/tree/integration).

**Comment by NodeGuy** [comment]
Implemented in e0d2162dc98ec2736c5d87cebc4b4653b41125be.
